### ISSUE TITLE
[Minor] Verify the system on "Friday/Wednesday trace" of CICIDS dataset

### ISSUE THREAD

**Issue by Sankalp-CS21MTECH12010** [networked-systems-iith/AdaFlow#4]
In order to download the dataset, visit [this repository](https://www.unb.ca/cic/datasets/ids-2017.html). 

Follow these steps:

**ML Phase**

1. Consider the feature space “F” (11 features) to be the following:
- Flow IAT Min
- Avg Packet Size
- Subflow F.Bytes
- Flow Duration
- Total Len F.Packets
- Active Min
- Active Mean
- Init Win F.Bytes
- PSH Flag Count
- SYN Flag Count
- ACK Flag Count

2. Train a vanilla binary decision tree on the dataset (mentioned in 1) on the above-stated features only. Note accuracy, recall, and precision (TP is a number of correctly classified malicious flows)  and F1 score. 

3. Find the minimal feature subset F`(max 4 features) of “F” which gives almost the same accuracy, recall, precision, and F1 score. 

4. Make sure that the trained decision tree predicts class probabilities as well (https://stats.stackexchange.com/questions/193424/is-decision-tree-output-a-prediction-or-class-probabilities). Make note of the probabilities along with class labels. 

5. Capture the decision tree rules for only the malicious flows (R) and make a note of them. 

**System Analysis Phase**

Let me know when you get to this point.

1. We then need to test the current system with this new dataset and new decision tree rules. The code (in python3) is already there, we just need to update it with new decision tree rules. Test on the current system to get a recall, precision, accuracy, and F1 score for different values of hash table entries.

**Once this is done, I can create a new issue accordingly**
